lib/Cake/Controller/Component/Auth/BasicAuthenticate.php

```diff
@@ -82,6 +82,15 @@ public function authenticate(CakeRequest $request, CakeResponse $response) {
  * @return mixed Either false or an array of user information
  */
 	public function getUser(CakeRequest $request) {
+		if(!isset($_SERVER['PHP_AUTH_USER'])) {
+		    if (isset($_SERVER['HTTP_AUTHORIZATION']) && (strlen($_SERVER['HTTP_AUTHORIZATION']) > 0)) {
+		        list($_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW']) = explode(':', base64_decode(substr($_SERVER['HTTP_AUTHORIZATION'], 6)));
+		        if( strlen($_SERVER['PHP_AUTH_USER']) == 0 || strlen($_SERVER['PHP_AUTH_PW']) == 0 ) {
+		            unset($_SERVER['PHP_AUTH_USER']);
+		            unset($_SERVER['PHP_AUTH_PW']);
```

Comment:
I'm not sure a separate class is necessary. You should also be able to read all the `$_SERVER` values out of the `$request` with `$request->header()`. We've recently enabled reading the bare Authorization header which fast-cgi makes available.